### ISSUE TITLE
Batch processing with re-submit when validation errors

### ISSUE THREAD

**Issue by thonspeciebio** [instructor-ai/instructor#1092]
I have a complex input file of semi structured data that I want to re-structure using instructor. It does a great job with the direct chat completion calls, returning my Parsed Pydantic model just fine.

But when I create a BatchJob, it seems that it simply runs one job and let Pydantic model validate the model and just returns the failed instances, but does not attempt to re-ask the job, sending it back as another job.

Are there any plans to implement re-ask mechanisms for batch jobs? Batch jobs are soo much faster and cheaper for me, but the pydantic validation errors are holding me back...

Maybe someone has some hints for me on re-using some of the code already used for the regular chat completion?

**Comment by jxnl** [comment]
how would that work, 30 out of 500 fail, do you want us to then make 30 more requests when you load and notice they failed? 30 requests now, or submit a second batch job?

**Comment by thonspeciebio** [comment]
Yup, exactly how I have implemented it now myself. I collect the invalid lines from the BatchJob and re-submit those until they work (or when max-try is exhaustive.

**Comment by jxnl** [comment]
Closing as part of repository maintenance for issues created before 2025.
